app/controllers/api/cloud_templates_controller.rb

```diff
@@ -8,6 +8,35 @@ def check_compliance_resource(type, id, _data = nil)
       end
     end
 
+    def import_resource(_type, _id, data = {})
+      params = %w(dst_provider_id src_provider_id src_image_id cos_provider_id cos_bucket_id)
+      raise BadRequestError, "Parameter 'data' has to contain non-empty values for the keys '#{params.join(", ")}', received: '#{data}'" if data.values_at(*params).any?(&:blank?)
+      raise BadRequestError, "Source and destination provider identifiers must differ"  if data['dst_provider_id'] == data['src_provider_id']
+
+      ems_dst = resource_search(data['dst_provider_id'], :providers, collection_class(:providers))
+      raise BadRequestError, "Specified destination provider cannot be found by this id: '#{data['dst_provider_id']}'" if ems_dst.nil?
+
+      ems_src = resource_search(data['src_provider_id'], :providers, collection_class(:providers))
+      raise BadRequestError, "Specified source provider cannot be found by this id: '#{data['src_provider_id']}'" if ems_src.nil?
+
+      src_image = resource_search(data['src_image_id'], :providers, collection_class(:templates))
```

Comment:
@iv1111 We'll need to check that `src_image.supports?(:import_image)` here

Reply:
You probably mean _ems_src_. It's the properties of source and destination providers to support import.

Right now I'm not sure how to implement this properly. It has to meet at both ends. Namely, instance denoted by _ems_dst_ has to support import from the instance denoted as _ems_src_. E.g. hypothetical situation - PVS supports import from PVC but not OpenStack, while OpenStack supports import from PVS (i.e. a directed graph).

Also something to note. Support of import can't be defined fixed for all instances of a given class. It depends on whether, say, user has done some preparations on the PowerVC server, prior to caling this API method.

Reply:
If possible, It'd be great if we could postpone the validation of this particular case and create a work item for it not to forget, mentioning it in this PR.

Reply:
I think it's more like we have to do something like this:

`ok = dst_provider.supports_import_from?(src_provider_id) && src_provider.supports_export_to?(dst_provider_id)`

This way one can be sure that the _dst_provider_ has done manual or any other preparations (if any) to support the importing from this particular _src_provider_ type and that _src_provider_ is capable of providing the image to _dst_provider_. I guess that doing it this way would also defy the purpose for defining a graph using "allowed_*" .

However, in such case _supports_import_from_ and _supports_export_to_ would have to check for class names and set _supports_ value accordingly, which isn't that nice.

Reply:
Imagine PVS with id = 1 generally supports import from PVC but this particular PVC with id=6  isn't prepared for that (and not even menat to), while PVC instance with id = 8 is prepared. So having a statically defined graph somewhere doesn't help you to determine if the execution can continue.

Reply:
If you have to check that a provider is configured to import from/export to a specific other provider then `supports feature` is not the right way to do this.

Reply:
Okay well the API isn't the right place to be doing this, we are going to need to look at filtering down what are "eligible" selections when building the dialog dropdowns the API should just take what it is given and pass it along.